### DISCUSSION TITLE
Problems with the number imput in combination with react-hook-form

### DISCUSSION THREAD

**Discussion by manuelpoelzl** [react-hook-form/react-hook-form#8158] [Q&A] [answered]
Hey guys,

I have a problem with the number input in combination with react-hook-form.
Currently I use the components of PrimeReact but also with the native <input type="number"/> the problem occurs.

Specifically, I can't enter any values, as soon as I specify something the input field shows NaN. But for all other fields, like checkboxes or text fields, the values can be changed without any problems.

I am currently using the controller from react-hook-form, which is also shown in the PrimeReact documentation.

I would be very grateful if someone here could point me in the right direction 😉

**Comment by Moshyfawn** [comment]
Please, consider providing a reproduction codesanbox

**Reply by manuelpoelzl** [reply]
Here you go,
I've added three fields here, a text field, a dropdown, and a number input. 
All fields except the Number field behave as they should, but as soon as you type something into the Number field NaN appears.

https://codesandbox.io/s/primereact-test-forked-l336rb?file=/src/index.js:2440-2444

**Comment by Moshyfawn** [comment, marked as answer]
According to [the PrimeReact `InputNumber` API docs](https://www.primefaces.org/primereact/inputnumber/), the input returns a custom `event` object `onChange` that contains `originalEvent` and `value`.

By spreading `{...field}`, you're ecentially passing the whole `event` object into the field's handler, which results in the field's value being an `event` object.

As it's a controlled input _(using `Controller` API and passing the `NumberInput` it's own value back)_, you set the input's value to be the `event` object. The `NumberInput` tries to parse the value into a `number`, but fails with `NaN`, as that's the result of trying to convert an `object` to `number`.

So, at the end of the day, it's not a problem with RHF, but with how you update the field's value.
```diff
- {...field}
+ onChange={event => field.onChange(event.value)}
```

**Reply by manuelpoelzl** [reply]
Ah I see, that makes sense then.
Thanks for the explanation!
